cdap-docs/developers-manual/source/data-exploration/filesets.rst

```diff
@@ -0,0 +1,86 @@
+
+Explore Properties
+------------------
+A ``FileSet`` or ``TimePartitionedFileSet`` can be made explorable by setting several properties when
+creating the dataset. For example, in the configure method of your application::
+
+    Schema schema = Schema.recordOf(
+      "purchase",
+      Schema.Field.of("user", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("item_id", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("price", Schema.of(Schema.Type.DOUBLE))
+    );
+    createDataset("my-files", "fileSet", FileSetProperties.builder()
+      .setBasePath("mylocation")
+      .setInputFormat(AvroKeyInputFormat.class)
+      .setOutputFormat(AvroKeyOutputFormat.class)
+      // everything past here is an explore property
+      .setEnableExploreOnCreate(true)
+      .setSerDe("org.apache.hadoop.hive.serde2.avro.AvroSerDe")
+      .setExploreInputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat")
+      .setExploreOutputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat")
+      .setTableProperty("avro.schema.literal", schema.toString())
+      .build());
+
+These Dataset properties map directly to table properties in Hive. 
+In fact, the create table statement that will be generated for the table above is::
+
+  CREATE EXTERNAL TABLE cdap_user_myfiles(
+    user string,
+    item_id int,
+    price double)
+  ROW FORMAT SERDE "org.apache.hadoop.hive.serde2.avro.AvroSerDe"
+  STORED AS INPUTFORMAT "org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat"
+  OUTPUTFORMAT "org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat"
+  LOCATION "/cdap/mylocation"
+  TBLPROPERTIES (
+    "avro.schema.literal"="{\"type\": \"record\", \"name\": \"stringBody\", \"fields\": [{ \"name\":\"ts\", \"type\":\"long\" }, { \"name\":\"body\", \"type\":\"string\" } ] }"
+  );
+
+Please see the `Hive Language Manual
+<https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-Create/Drop/TruncateTable>`__
+for more information.
+
+Limitations
+-----------
+As exploration is tied directly to Hive, file exploration has several limitations that mirror limitations in Hive.
+
+- All explorable files must be of the same format.
+- All explorable files must be in a format known to the version of Hive you are using.
+- Some versions of Hive may try to create a temporary staging directory at the table location when executing queries.
+  If you are seeing permissions errors, try setting ``hive.exec.stagingdir`` in your Hive configuration to ``/tmp/hive-staging``.
+
+A ``FileSet`` has some additional limitations that the ``TimePartitionedFileSet`` does not have.
+
+- Hive tables created are not partitioned. This means any query will perform a full table scan.
+- Only files at the base location of the ``FileSet`` are visible to queries. Directories are not read.
+  Since Mapreduce writes output files to a directory, if you want Mapreduce output to be explorable you
```

Comment:
Since MapReduce writes output files to a directory, you must move all output files to the base location for MapReduce output to be explorable.